### ISSUE TITLE
Maintenance mode for the cephcsi resources

### ISSUE THREAD

**Issue by Madhu-1** [ceph/ceph-csi-operator#75]
# Describe the feature you'd like to have #

If there is a change in the cephcsi artifacts like deployment/daemonset/service/csidriver object, the operator should not reconcile it if certain annotation is set

# Who is the end user and what is the use case where this feature will be valuable?
  
This is useful for development and also useful for the customers whey they want to make changes to the deployment in some cases but want operator not to revert it back

# How will we know we have a good solution? (acceptance criteria) #

Add a list of criteria that should be met for this feature to be useful

# Additional context #

**Comment by travisn** [comment]
Would it be sufficient to just scale down the csi operator during maintenance?

**Comment by Madhu-1** [comment]
> Would it be sufficient to just scale down the csi operator during maintenance?

@travisn yes that work but not for all the cases like the admin wants to make one manual changes to the deployment (tuning certain configurations which are not configurable) but wants to keep the operator running to ensure that all the operator functionality works (like csi configmap update etc)
